baybe/kernels/base.py

```diff
@@ -12,31 +14,60 @@
     unstructure_base,
 )
 from baybe.serialization.mixin import SerialMixin
-from baybe.utils.basic import filter_attributes
+from baybe.utils.basic import filter_attributes, get_parent_classes
+
+if TYPE_CHECKING:
+    import torch
 
 
 @define(frozen=True)
 class Kernel(ABC, SerialMixin):
     """Abstract base class for all kernels."""
 
-    lengthscale_prior: Optional[Prior] = field(default=None, kw_only=True)
-    """An optional prior on the kernel lengthscale."""
-
-    def to_gpytorch(self, *args, **kwargs):
+    def to_gpytorch(
```

Comment:
@Scienfitz, @AVHopp: I refactored the logic a bit to make the arguments explicit (i.e. remove args/kwargs) which seems a bit clearer/safer to me. Please let me know if this makes sense to you

Reply:
Hm, weird. Coming from my code, even though I currently have no clue what might have caused this 😄

Reply:
Keeping this open as a reminder to check the mypy error at the end.

Reply:
It seems like accessing the `__init__` method of a `type` is considered unsafe in general, see the discussion here: https://github.com/python/typing/discussions/1331 However, since we do not execute that, I think we can safely ignore the error, or what is your opinion here?

Reply:
Have simply deactivated since we already agreed that the problematic line of code is going to be replaced in an upcoming PR anyway 👍🏼